### ISSUE TITLE
Build failure when linking DUB dependencies with long names

### ISSUE THREAD

**Issue by mbierlee** [dlang/dub#2538]
### System information
- **dub version**: DUB version 1.30.0, built on Nov 14 2022
- **OS Platform and distribution**: Windows 10
- **compiler version** DMD32 D Compiler v2.101.0-dirty

### Bug Description
When adding a dub project with a very long name, a linker error occurs. In the following example it tries to build a dependency called `poodinis-mirage-config-injector`:
```
Build directory C:\Users\[..]\AppData\Local\dub\packages\poodinis-mirage-config-injector-1.0.1\poodinis-mirage-config-injector\.dub\build\library-debug-windows-x86_64-dmd_v2.101.0-dirty-AD9F674B70EEEA986AF59B29018C0DC50245C96E020D599F9A3615568E3B78BA\ is not writable. Falling back to direct build in the system's temp folder.
    Building poodinis-mirage-config-injector 1.0.1 [library]
    Building dublongpaths ~master: building configuration [application]
     Linking dublongpaths
lld-link: error: undefined symbol: _D8poodinis13valueinjector6mirage10loadConfigFOCQBu9container19DependencyContainerxAyaZv
>>> referenced by D:\Temp\dublongpaths\source\app.d:22
>>>               .dub\build\application-debug-windows-x86_64-dmd_v2.101.0-dirty-BEE0070E13F5C7E5928CD6E33EABF03EF415566C2C52A4FE23C21BB581FD23B2\dublongpaths.obj:(_Dmain)

lld-link: error: undefined symbol: _D8poodinis13valueinjector6mirage12__ModuleInfoZ
>>> referenced by .dub\build\application-debug-windows-x86_64-dmd_v2.101.0-dirty-BEE0070E13F5C7E5928CD6E33EABF03EF415566C2C52A4FE23C21BB581FD23B2\dublongpaths.obj:(_D3app12__ModuleInfoZ)
Error: linker exited with status 1
Error D:\DLang\dmd2\windows\bin\dmd.exe failed with exit code 1.
```

This is possibly related to issue #911. The dependency fails to compile so the linker fails to find the lib.

### How to reproduce?
See project: [dublongpaths.zip](https://github.com/dlang/dub/files/10099155/dublongpaths.zip)
Just `dub run` it.

**Comment by mbierlee** [comment]
The problem lies in the `writeFile` function in https://github.com/dlang/dub/blob/abe51f78ff23b0f85b8a5c22be08c45b179637f1/source/dub/internal/vibecompat/core/file.d (Line 26). It fails to write the test file from https://github.com/dlang/dub/blob/4ff336e2c7371b739c5d4d7088c1d85d75da4d0f/source/dub/generators/build.d (Line 264)

One possible fix it to prefix the path with `\\?\`. See also: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

**Comment by mbierlee** [comment]
I have since renamed the project to be shorted. The attached example project might not work as-is anymore.

**Comment by WebFreak001** [comment]
fixed by #2589
